File: desktop/src/shared/api/diag.ts
import { invokeTauri } from "@/shared/api/tauri";

/**
 * Temporary diagnostics tap (feat/observer-diag): forwards gate-level
 * observer diagnostics to the tauri process stdout, where a terminal
 * launch captures them. Fire-and-forget; never throws into callers.
 */
export function diag(message: string): void {
  void invokeTauri("diag_log", { message }).catch(() => {});
}

File: desktop/src-tauri/src/commands/diag.rs
use tauri::command;

/// Temporary diagnostics tap (feat/observer-diag): the webview has no
/// readable console in release builds, so gate-level observer diagnostics
/// are routed here and printed to stdout, where a terminal launch of the
/// app captures them. Remove with the branch once the observer-frame drop
/// is diagnosed.
#[command]
pub fn diag_log(message: String) {
    println!("[diag] {message}");
}
